File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/storage/WaypointStorageImplementation.java
package de.cubbossa.pathfinder.storage;

import de.cubbossa.pathfinder.node.implementation.Waypoint;
import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

public interface WaypointStorageImplementation {

  Optional<Waypoint> loadWaypoint(UUID uuid);

  Collection<Waypoint> loadWaypoints(Collection<UUID> ids);

  Collection<Waypoint> loadAllWaypoints();

  void saveWaypoint(Waypoint node);

  void deleteWaypoints(Collection<Waypoint> waypoints);

}

File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/storage/WaypointStorageImplementation.kt
package de.cubbossa.pathfinder.storage

import de.cubbossa.pathfinder.node.implementation.Waypoint
import java.util.*

interface WaypointStorageImplementation {

    fun loadWaypoint(uuid: UUID): Waypoint?

    fun loadWaypoints(ids: Collection<UUID>): Collection<Waypoint>

    fun loadAllWaypoints(): Collection<Waypoint>

    fun saveWaypoint(node: Waypoint)

    fun deleteWaypoints(waypoints: Collection<Waypoint>)
}
